qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -254,11 +254,23 @@ def _draw_widgets(self):
     @pyqtSlot()
     def maybe_hide(self):
         """Hide the statusbar if it's configured to do so."""
+        strategy = config.val.statusbar.show
         tab = self._current_tab()
-        hide = config.val.statusbar.hide
-        if hide or (tab is not None and tab.data.fullscreen):
+        if tab is not None and tab.data.fullscreen:
             self.hide()
-        else:
+        elif strategy == 'never':
+            self.hide()
+        elif strategy == 'in_mode':
+            try:
+                mode_manager = modeman.instance(self._win_id)
+                if mode_manager.mode == usertypes.KeyMode.normal:
+                    self.hide()
+                else:
+                    self.show()
+            except KeyError:
```

Comment:
This seems a bit out of place, because it's a very generic exception. Maybe `modeman.instance(...)` could catch the `KeyError` and raise a custom `modeman.UnavailableError` exception which then can be handled here? Then the code would also be clear enough without the comment.

Also, please only have as little as possible in the `try:` block:

```python3
try:
    mode_manager = modeman.instance(self._win_id)
except modeman.UnavailableError:
    self.hide()
else:
    if ...:
        ...
```

Reply:
I have modified `modeman.instance(..)` to raise an `UnavailableError` by specifing `None` as a default return value for `objrej.get(..)`, and checking whether the value returned is in fact `None` or not. I think this is a good strategy, but I'm not 100% sure either.